### ISSUE TITLE
Add T8 series models

### ISSUE THREAD

**Issue by brcros** [bmartin5692/bumper#108]
Hello, great work. 
What about to add OZMO T8 series into the model ?
Thanks.

**Comment by rickyn172** [comment]
Would this library help with spending up T8 support? 
https://github.com/mrbungle64/ecovacs-deebot.js

**Comment by edenhaus** [comment]
In #116 I updated the models, which also includes the T8.
But no guarantee that it will work as I only have the Ozmo 950.

**Comment by AndreCox** [comment]
I have an osmo t8 if any testing is required I'll gladly do so

**Comment by AndreCox** [comment]
> Would this library help with spending up T8 support?
> https://github.com/mrbungle64/ecovacs-deebot.js

I don't think so as this library just allows you to control a T8 through the original servers, anyone adding support for the T8 would need to reverse engineer the code running on the servers, however a lot of work has already been done so I shall have a look at my ozmo T8 and see if I can capture some of the commands being sent and received. If I had to guess I would assume it is using the XMPP protocol.

**Comment by edenhaus** [comment]
> I don't think so as this library just allows you to control a T8 through the original servers, anyone adding support for the T8 would need to reverse engineer the code running on the servers, however a lot of work has already been done so I shall have a look at my ozmo T8 and see if I can capture some of the commands being sent and received. If I had to guess I would assume it is using the XMPP protocol.

The T8 uses MQTT/JSON. All new robots will probably use it.

As my PR is merge, can someone test it if the T8 is now working?

**Comment by AndreCox** [comment]
> > I don't think so as this library just allows you to control a T8 through the original servers, anyone adding support for the T8 would need to reverse engineer the code running on the servers, however a lot of work has already been done so I shall have a look at my ozmo T8 and see if I can capture some of the commands being sent and received. If I had to guess I would assume it is using the XMPP protocol.
> 
> The T8 uses MQTT/JSON. All new robots will probably use it.
> 
> As my PR is merge, can someone test it if the T8 is now working?

Yes I'll get it set up and do some testing, anything I should be looking out for that might not work?

**Comment by mjdyson** [comment]
Hello. I'm a bit of a novice on the coding side, but I have the T8 AIVI working. However, with a caveat - the Ecovacs app has connection issues and does not connect to Bumper or the Bot.  I use DNS forwarding on my router to Bumper running in Docker  

To control the vacuum, I use an integration with HomeAssistant called "Deebot for Home Assistant". All the documentation doesnt seem to be up to date, or include any mention of Bumper so it was a bit of a struggle to get it working.

**Comment by AndreCox** [comment]
> Hello. I'm a bit of a novice on the coding side, but I have the T8 AIVI working. However, with a caveat - the Ecovacs app has connection issues and does not connect to Bumper or the Bot.  I use DNS forwarding on my router to Bumper running in Docker  
> 
> To control the vacuum, I use an integration with HomeAssistant called "Deebot for Home Assistant". All the documentation doesnt seem to be up to date, or include any mention of Bumper so it was a bit of a struggle to get it working.
> 

Probably because you need to set the dns in your phone or it will default to 8.8.8.8 and 8.8.4.4, I'm assuming your on Android

**Comment by mjdyson** [comment]
You've hit on something there with the DNS (Android 8.1 Lineage OS tablet - it's old)! I've manually set the DNS in the wifi settings of the tablet to my router (as Primary and Secondary DNS), but I dont think it's picking it up properly.

When I'm on other devices, if I head to portal-eu.ecouser.net it normally takes me to a (self hosted) Bumper page - presumably as a consequence of the DNS forward at the router. When I try to do this from the Android tablet, it wont load anything. I've a feeling this is the problem. I can however, go directly to the IP address of the Bumper server (port 8007) where it loads the same Bumper page, so there doesnt seem to be anything wrong / blocking the network. I thought for a minute there could be something wrong with cross vlan rules I've set up.

**Comment by Robert-D** [comment]
Hey guys, I am also trying to get Bumper to work with my T8. I have setup the docker container according to the documentation and have configured AdGuard to rewrite the 3 domains. I see my robot querying mq-ww.ecouser.net and AdGuard is responding with the Bumper server address, however I don't see robot showing up under the Bots table in Bumper. Has it been confirmed to work with the T8 models yet?

**Comment by mjdyson** [comment]
Hi @Robert-D - I have my T8 AIVI working, yes. 
Wrt the DNS entries, it might take a reconnect for your vac to pick up the latest options. Take the vacuum off the charger, turn it off (with the red switch under the lid) and wait a few sec and then switch it back on.

If I go to https://portal-eu.ecouser.net/ it redirects to my local server that shows some tables with the vacuum and MQTT users etc. The vac appears in these lists along with my controllers (clients). I use the HomeAssistant software with an integration to interact with it. It works really well and I can script cleaning of individual zones quite easily.
As for the Android app, I've given up on getting that to work. The DNS over https is a battle I just cant figure out - I'm assuming that the router cant inspect it, so it fails to resolve properly.

Edit: My DNS redirect is done directly on my router (Unifi USG)

**Comment by Robert-D** [comment]
> Hi @Robert-D - I have my T8 AIVI working, yes.
> Wrt the DNS entries, it might take a reconnect for your vac to pick up the latest options. Take the vacuum off the charger, turn it off (with the red switch under the lid) and wait a few sec and then switch it back on.
> 
> If I go to https://portal-eu.ecouser.net/ it redirects to my local server that shows some tables with the vacuum and MQTT users etc. The vac appears in these lists along with my controllers (clients). I use the HomeAssistant software with an integration to interact with it. It works really well and I can script cleaning of individual zones quite easily.
> As for the Android app, I've given up on getting that to work. The DNS over https is a battle I just cant figure out - I'm assuming that the router cant inspect it, so it fails to resolve properly.
> 
> Edit: My DNS redirect is done directly on my router (Unifi USG)

Thanks for your reply. Does your bot show up under the "Bots" section of bumper or only the MQTT section?

**Comment by mjdyson** [comment]
It's in both :)

**Comment by edenhaus** [comment]
@mjdyson As you have a T8 AIVI, is the camera live stream feature working with bumper?

**Comment by mjdyson** [comment]
I'm not sure @edenhaus - is it possible to test without the Android app? I mean, is there a local URL for the camera stream that I could open?

**Comment by edenhaus** [comment]
I don‘t know it as I only have the 950. But as the T9 AIVI was released today, a friend asked me about it and I forwared the question to you.


> To control the vacuum, I use an integration with HomeAssistant called "Deebot for Home Assistant". All the documentation doesnt seem to be up to date, or include any mention of Bumper so it was a bit of a struggle to get it working.

I know that the docs of the integration is bad, but I currently don‘t have time to improve it. But we plan to start improve it soon.

Have you iOS devices? There is it quite easy to get the app running. On android you need to have the device rooted and afterwards you can add custom certificates

**Comment by mjdyson** [comment]
Yeah, I played with the certificate a few months back on a rooted Nexus 9 tablet. It seemed to be stored in the correct certificate settings area, but the app struggled to load the vacuum. The live map works pretty well through the integration and I've got some scripts working that enable room by room cleaning on demand. Thank you for the work on that (happy to share details).

As for the iOS testing, sure.. it's getting late here, so I'll see if I can get it working on an old iPad the kids use tomorrow. Will report back.

**Comment by edenhaus** [comment]
Sharing and improvements (also only the docs) are always welcome 😊 Please open a PR (or issue) directly in "Deebot for Home Assistant".

No stress😊Don‘t forget to trust the certificate after adding it on iOS, took me some time to find out, that you need to trust them after importing

**Comment by pcdiks** [comment]
> @mjdyson As you have a T8 AIVI, is the camera live stream feature working with bumper?

I have bumper working with my Ecovacs T8 AIVI but the camera is not working.

**Comment by edenhaus** [comment]
@pcdiks The camera cannot be working as this functionality is missing. I was only interested if the camera stream is really encrypted or is ecovacs only saying that

**Comment by MitchellHayes** [comment]
Hoping to get this set up at home, I have a T9+ what is the status of functionality with this vacuum?
